### PR TITLE
Turned warnings-as-errors on by default.

### PR DESCRIPTION
**Pull request by AWoloszyn**
Fixed a few warnings that appear in builds on VS2013 and VS2015.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -43,6 +43,7 @@ if ("${CMAKE_BUILD_TYPE}" STREQUAL "")
   set(CMAKE_BUILD_TYPE "Debug")
 endif()
 
+option(SPIRV_WERROR "Enable error on warning" ON)
```
Comment (dneto0) [inline CMakeLists.txt]: Let's document in the README

Comment (dneto0) [inline CMakeLists.txt]: I mean update the existing description in the readme.

**Comment by dneto0** [comment]
-1 just to update the README.
I reviewed all the places where this patch adds uint16_t(  )  constructors. They are all bounds checked either explicitly, or by construction from surrounding code.

File: CMakeLists.txt (hunk 1 above)
Comment (AWoloszyn) [inline CMakeLists.txt]: Done.

**Comment by antiagainst** [comment]
+1

**Comment by dneto0** [comment]
+2

**Comment by AWoloszyn** [comment]
Merged manually.

**Pull request closed without merging**